--- linphone-3.5.1/gtk/setupwizard.c	2012-02-08 16:57:35.000000000 +0200
+++ linphone-3.5.1/gtk/setupwizard.c	2012-02-19 19:51:01.487146154 +0200
@@ -270,7 +270,7 @@
 	linphone_account_creator_set_username(creator, gtk_entry_get_text(username));
 
 	if (g_regex_match_simple("^[a-zA-Z]+[a-zA-Z0-9.\\-_]{3,}$", gtk_entry_get_text(username), 0, 0)) {
-		g_thread_create(check_username_availability, (void*)w, FALSE, NULL);
+		g_thread_new (NULL, check_username_availability, (void*)w);
 	}
 	else {
 		if (gtk_entry_get_text_length(username) < LOGIN_MIN_SIZE) {
@@ -534,7 +534,7 @@
 	ok = create_pixbuf(linphone_gtk_get_ui_config("ok","ok.png"));
 	notok = create_pixbuf(linphone_gtk_get_ui_config("notok","notok.png"));
 
-	g_thread_init (NULL);
+	g_type_init ();
 	gdk_threads_init ();
 
 	GtkWidget *p1=create_intro();
