--- linphone-1.2.0/mediastreamer/audiostream.c.maxrtp	2006-01-03 04:26:06.000000000 -0500
+++ linphone-1.2.0/mediastreamer/audiostream.c	2006-01-03 04:26:28.000000000 -0500
@@ -112,7 +112,7 @@
 			RtpSession **recvsend){
 	RtpSession *rtpr;
 	rtpr=rtp_session_new(RTP_SESSION_SENDRECV);
-	rtp_session_max_buf_size_set(rtpr,MAX_RTP_SIZE);
+/*	rtp_session_max_buf_size_set(rtpr,MAX_RTP_SIZE);*/
 	rtp_session_set_profile(rtpr,profile);
 	rtp_session_set_local_addr(rtpr,get_local_addr_for(remip),locport);
 	if (remport>0) rtp_session_set_remote_addr(rtpr,remip,remport);
@@ -133,7 +133,7 @@
 	/* creates two rtp filters to recv send streams (remote part)*/
 	
 	rtps=rtp_session_new(RTP_SESSION_SENDONLY);
-	rtp_session_max_buf_size_set(rtps,MAX_RTP_SIZE);
+/*	rtp_session_max_buf_size_set(rtps,MAX_RTP_SIZE);*/
 	rtp_session_set_profile(rtps,profile);
 #ifdef INET6
 	rtp_session_set_local_addr(rtps,"::",locport+2);
@@ -147,7 +147,7 @@
 	rtp_session_set_jitter_compensation(rtps,jitt_comp);
 	
 	rtpr=rtp_session_new(RTP_SESSION_RECVONLY);
-	rtp_session_max_buf_size_set(rtpr,MAX_RTP_SIZE);
+/*	rtp_session_max_buf_size_set(rtpr,MAX_RTP_SIZE);*/
 	rtp_session_set_profile(rtpr,profile);
 #ifdef INET6
 	rtp_session_set_local_addr(rtpr,"::",locport);
