diff -up linphone-3.6.1/mediastreamer2/src/videofilters/theora.c.fix linphone-3.6.1/mediastreamer2/src/videofilters/theora.c
--- linphone-3.6.1/mediastreamer2/src/videofilters/theora.c.fix	2014-11-14 14:34:59.416313134 -0500
+++ linphone-3.6.1/mediastreamer2/src/videofilters/theora.c	2014-11-14 14:35:44.657985230 -0500
@@ -332,7 +332,7 @@ static void enc_process(MSFilter *f){
 		if (need_send_conf(s,elapsed)){
 			if (s->packed_conf) {
 				om=dupmsg(s->packed_conf);
-				ms_message("sending theora packed conf (%i bytes)",msgdsize(om));
+				ms_message("sending theora packed conf (%i bytes)",(int)msgdsize(om));
 				packetize_and_send(f,s,om,timestamp,THEORA_PACKED_CONF);
 			}else {
 				ms_error("No packed conf to send.");
