diff -Naur linphone-2.1.1.orig/oRTP/Makefile.am linphone-2.1.1/oRTP/Makefile.am
--- linphone-2.1.1.orig/oRTP/Makefile.am	2007-10-18 22:05:10.000000000 +0200
+++ linphone-2.1.1/oRTP/Makefile.am	2009-03-13 08:38:34.000000000 +0100
@@ -12,14 +12,13 @@
 if HAVE_DOXYGEN
 
 # doxdir & pkgdocdir are not always defined by automake
-docdir=$(datadir)/doc
 pkgdocdir=$(docdir)/$(PACKAGE)-$(VERSION)
 doc_htmldir=$(pkgdocdir)/html
 
 doc_html_DATA = $(top_builddir)/doc/html/html.tar
 
 $(doc_html_DATA): $(top_builddir)/doc/html/index.html
-	cd $(<D) && tar cf html.tar *
+	cd $(top_builddir)/doc/html && tar cf html.tar *
 
 $(top_builddir)/doc/html/index.html: $(SOURCES) ortp.doxygen Makefile.am
 	rm -rf doc
@@ -51,7 +50,7 @@
 .PHONY: package
 
 PKG_NAME = $(PACKAGE)-$(VERSION)-$(RELEASE)
-BUILDROOT = $(shell pwd)/epm-install
+BUILDROOT = `pwd`/epm-install
 
 package: $(srcdir)/pkg.list $(srcdir)/configure
 	-rm -rf pkg $(BUILDROOT) $(PKG_NAME).*
