From 3873e40dc1c3c04d76ae22bf1169cecec5193624 Mon Sep 17 00:00:00 2001
From: Simon Morlat <simon.morlat@linphone.org>
Date: Mon, 04 Apr 2011 22:38:42 +0000
Subject: restore intl of .ui files (thanks intltool grrr)

---
diff --git a/gtk/main.c b/gtk/main.c
index e73ae8a..5e16710 100644
--- a/gtk/main.c
+++ b/gtk/main.c
@@ -1431,8 +1431,11 @@ int main(int argc, char *argv[]){
 	g_object_set(settings, "gtk-menu-images", TRUE, NULL);
 	g_object_set(settings, "gtk-button-images", TRUE, NULL);
 
-	if (workingdir!=NULL)
-		chdir(workingdir);
+	if (workingdir!=NULL){
+		if (chdir(workingdir)==-1){
+			g_error("Could not change directory to %s : %s",workingdir,strerror(errno));
+		}
+	}
 
 	/* Now, look for the factory configuration file, we do it this late
 		 since we want to have had time to change directory and to parse
