--- linphone-0.12.2/share/Makefile.in.orig	2005-03-23 16:26:52.922690626 -0500
+++ linphone-0.12.2/share/Makefile.in	2005-03-23 16:27:13.949932428 -0500
@@ -184,7 +184,7 @@
 confexdir = $(datadir)/linphonec
 confex_DATA = linphonec
 
-pkgconfigdir = $(prefix)/lib/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = linphone.pc
 
 EXTRA_DIST = $(LINPHONE_SOUNDS) \
--- linphone-0.12.2/share/linphone.pc.in.orig	2005-03-23 21:10:49.392763407 -0500
+++ linphone-0.12.2/share/linphone.pc.in	2005-03-23 21:11:21.983938389 -0500
@@ -1,7 +1,7 @@
-prefix=@prefix@
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
 
 
 Name: liblinphone
