--- linphone-0.12.2/share/linphone.pc.in.orig	2005-03-23 21:10:49.392763407 -0500
+++ linphone-0.12.2/share/linphone.pc.in	2005-03-23 21:11:21.983938389 -0500
@@ -1,7 +1,7 @@
-prefix=@prefix@
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
 
 
 Name: liblinphone
--- linphone-1.0.1/share/Makefile.in.pkgconfig	2005-03-23 03:17:48.000000000 -0500
+++ linphone-1.0.1/share/Makefile.in	2005-03-24 13:40:41.828557116 -0500
@@ -286,7 +286,7 @@
 #to be compliant with freedesktop.org:
 linphone_fddir = $(prefix)/share/applications/
 linphone_fd_DATA = linphone.desktop
-pkgconfigdir = $(prefix)/lib/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = linphone.pc
 EXTRA_DIST = $(LINPHONE_SOUNDS) \
 								$(LINPHONE_RINGS) \
